packages/dmn-js-shared/src/features/modeling/behavior/NameChangeBehavior.js

```diff
@@ -0,0 +1,87 @@
+import CommandInterceptor from 'diagram-js/lib/command/CommandInterceptor';
+
+import {
+  getBusinessObject,
+  is
+} from 'dmn-js-shared/lib/util/ModelUtil';
+
+
+export default class NameChangeBehavior extends CommandInterceptor {
+
+  static $inject = [ 'eventBus', 'modeling' ];
+
+  constructor(eventBus, modeling) {
+    super(eventBus);
+
+    this._modeling = modeling;
+
+    this.postExecuted('element.updateProperties', this.updateVariableFromElement);
+    this.postExecuted('element.updateLabel', this.updateVariableFromLabel);
+  }
+
+  updateVariableFromLabel = ({ context }) => {
+    const { element, newLabel } = context;
+    const bo = getBusinessObject(element);
+
+    if (!bo.variable) {
+      return;
+    }
+
+    bo.variable.name = newLabel;
+    this._modeling.updateProperties(element, { variable: bo.variable });
```

Comment:
This is a technique called monkey-patching. If you add tests for `commandStack.undo`, you will notice that the change cannot be undone. We need to dispatch an updateProperties call which updates the variable name.
@philippfromme perhaps it makes sense you look into this together with @abdul99ahad

Reply:
Thanks, @barmac, for pointing this out. I discussed the monkey patching technique with @philippfromme earlier and I didn't realize that my implementation went against the recommended approach. I apologize for the oversight. I've updated the implementation to align with the correct approach.